bip-0352.mediawiki

```diff
@@ -0,0 +1,476 @@
+
+Alice performs the tweak with the sum of her input private keys in the following manner:
+
+* Let ''outpoints_hash = hash(txid<sub>0</sub> || vout<sub>0</sub> || ... || txid<sub>n</sub> || vout<sub>n</sub>)''
+* Let ''a = a<sub>0</sub> + a<sub>1</sub> + ... + a<sub>n</sub>''
+* Let ''P<sub>0</sub> = B + hash(outpoints_hash·a·B || 0)·G''
+
+''' Spend and Scan Key '''
+
+Since Bob needs his private key ''b'' to check for incoming payments, this requires ''b'' to be exposed to an online device. To minimize the risks involved, Bob can instead publish an address of the form ''(B<sub>scan</sub>, B<sub>spend</sub>)''. This allows Bob to keep ''b<sub>spend</sub>'' in offline cold storage and perform the scanning with the public key ''B<sub>spend</sub>'' and private key ''b<sub>scan</sub>''. Alice performs the tweak using both of Bob's public keys in the following manner:
+
+* Let ''P<sub>0</sub> = B<sub>spend</sub> + hash(outpoints_hash·a·B<sub>scan</sub> || 0)·G''
+
+Bob detects this payment by calculating ''P<sub>0</sub> = B<sub>spend</sub> + hash(outpoints_hash·b<sub>scan</sub>·A)·G'' with his online device and can spend from his cold storage signing device using ''(b<sub>spend</sub> + hash(outpoints_hash·b<sub>scan</sub>·A)) mod p'' as the private key.
+
+''' Labels '''
+
+For a single silent payment address of the form ''(B<sub>scan</sub>, B<sub>spend</sub>)'', Bob may wish to differentiate incoming payments. Naively, Bob could publish multiple silent payment addresses, but this would require him to scan for each one, which becomes prohibitively expensive. Instead, Bob can label his spend public key ''B<sub>spend</sub>'' with an integer ''m'' in the following way:
```

Comment:
In this section I think it would be also good to note that the motivation to implement labels basically as the simplest possible derivation scheme is easier scanning, right?

Aside from the `k++` note it may be also interesting to mention that given two different SP addresses with the same spend key the users could calculate the "label diff". That may give some hints on the usage of labels that is interesting for attackers.

It is also possible to spam labels if incremented integers are used as recemmended, right? E.g. given B0 I can calculate B1, B2 etc. myself by just adding `G` and send funds there to see what happens with them. This makes the label unreliable as the identifier of the source of the incoming payment.

Reply:
Point taken on the confusion around the term "silent payment address".

And I believe we've addressed the concerns around labels in the latest version of the BIP by making each label a unique 32 byte value.

Reply:
Hm, I don't think I see that change or maybe I am misunderstanding it? Currently, the BIP version here still talks about m as an "incremental integer starting from 1".

Reply:
The integer is still there but what has changed is that it's now hashed together with a secret. The label is no longer equivalent to `m` but it's now `hash(bscan || m)` where `bscan` and `m` are never revealed to outside observers. This makes it deterministic to recipients while indistinguishable from random to senders.

Reply:
Resolving this, let me know if you still have questions about this @fjahr